src/sparseml/exporters/transforms/quantize_qat_embedding.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from onnx import ModelProto, helper, numpy_helper
+
+from sparseml.exporters.transforms import OnnxTransform
+from sparseml.exporters.transforms.utils import (
+    INITIALIZER_MATCH,
+    assert_node_type,
+    get_structural_matches,
+    optional_node,
+    quantize_array,
+)
+from sparseml.onnx.utils import ONNXGraph, remove_node_and_params_from_graph
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def quantize_embedding(match, model):
+    graph = ONNXGraph(model)
+
+    input_quantize_node = match.parents[1][1]
+    input_dequantize_node = match.parents[1][2]
+
+    # quantize embedding
+    embedding_initializer = graph.get_init_by_name(input_quantize_node.input[0])
+    scale_initializer = graph.get_init_by_name(input_quantize_node.input[1])
+    zero_point_initializer = graph.get_init_by_name(input_quantize_node.input[2])
+
+    # arrays from embedding initializer
+    embedding = numpy_helper.to_array(embedding_initializer)
+    scale = numpy_helper.to_array(scale_initializer)
+    zero_point = numpy_helper.to_array(zero_point_initializer)
+
+    embedding_quant = quantize_array(embedding, scale, zero_point, zero_point.dtype)
+    embedding_quant_initializer = numpy_helper.from_array(
+        embedding_quant, name=f"{embedding_initializer.name}_quant"
+    )
+
+    # update graph
+    model.graph.initializer.append(embedding_quant_initializer)
+    match.node.input[1] = embedding_quant_initializer.name
+
+    # detect QDQ block on output
+    children = match.children
+    qdq_output = False
+    if children:
+        if len(children[0]) == 2:
+            if assert_node_type(children[0][0], "QuantizeLinear") and assert_node_type(
+                children[0][1], "DequantizeLinear"
+            ):
+                qdq_output = True
```

Comment:
a couple notes here:
1. children will always be a list so you can remove the if children
2. our matching code should guarantee that `len(children[i]) == len(children_ops[i])`, so you can get rid of the second if branch
3. if the values in the list are not None, then they should be guaranteed to be the op types that you specify in children ops, so i think you can just check if both are not none

thoughts?
```suggestion
    opt_quant, opt_dequant = match.children
    qdq_output = opt_quant is not None and opt_dequant is not None
```